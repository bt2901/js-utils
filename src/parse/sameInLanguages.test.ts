import parseSameInLanguages from './sameInLanguages';

describe('parseSameInLanguages', () => {
  it.each([
    [''],
    ['cs sh~ bg~'],
    ['j z'],
    ['ru cs (sh)'],
    ['ru yu'],
    ['ru# yu* mk~'],
    ['rue cs yu'],
    ['ub~ z j'],
    ['uk z bg'],
    ['v csb hsb cs'],
    ['v pl csb'],
    ['v pl sk~'],
    ['v sk j~'],
    ['v z yu~'],
    ['z'],
    ['z~ sh~'],
  ])('should parse correctly in legacy mode: %j', (encoded) => {
    const report = parseSameInLanguages(encoded, true);
    expect(report.toString()).toMatchSnapshot();
    const secondary = parseSameInLanguages(report.toString(), false);
    expect(secondary).toEqual(report);
  });

  it.each([
    [''],
    ['!be+ cs# !ru+ uk+ pl*'],
    ['!v+ z#'],
    ['!bg+ cz~ mk+'],
    ['sh* sl+'],
  ])('should parse correctly in new mode: %j', (encoded) => {
    const report = parseSameInLanguages(encoded);
    expect(report.toString()).toMatchSnapshot();
  });
});
