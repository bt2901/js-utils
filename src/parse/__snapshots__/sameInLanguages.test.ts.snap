// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseSameInLanguages should parse correctly in legacy mode: "" 1`] = `""`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "cs sh~ bg~" 1`] = `"!bg~ !bs~ !cnr~ !cs+ !hr~ !sk+ !sr~"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "j z" 1`] = `"!bg+ !bs+ !cnr+ !cs+ !hr+ !mk+ !pl+ !sk+ !sl+ !sr+"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "ru cs (sh)" 1`] = `"!cs+ !ru+ !sk+"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "ru yu" 1`] = `"!bs+ !cnr+ !hr+ !ru+ !sl+ !sr+"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "ru# yu* mk~" 1`] = `"!bs* !cnr* !hr* !mk~ !ru# !sl* !sr*"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "rue cs yu" 1`] = `"!bs+ !cnr+ !cs+ !hr+ !rue+ !sk+ !sl+ !sr+"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "ub~ z j" 1`] = `"!be~ !bg+ !bs+ !cnr+ !cs+ !hr+ !mk+ !pl+ !sk+ !sl+ !sr+ !uk~"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "uk z bg" 1`] = `"!bg+ !cs+ !pl+ !sk+ !uk+"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "v csb hsb cs" 1`] = `"!be+ !cs+ !csb+ !hsb+ !ru+ !sk+ !uk+"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "v pl csb" 1`] = `"!be+ !csb+ !pl+ !ru+ !uk+"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "v pl sk~" 1`] = `"!be+ !pl+ !ru+ !sk~ !uk+"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "v sk j~" 1`] = `"!be+ !bg~ !bs~ !cnr~ !hr~ !mk~ !ru+ !sk+ !sl~ !sr~ !uk+"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "v z yu~" 1`] = `"!be+ !bs~ !cnr~ !cs+ !hr~ !pl+ !ru+ !sk+ !sl~ !sr~ !uk+"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "z" 1`] = `"!cs+ !pl+ !sk+"`;

exports[`parseSameInLanguages should parse correctly in legacy mode: "z~ sh~" 1`] = `"!bs~ !cnr~ !cs~ !hr~ !pl~ !sk~ !sr~"`;

exports[`parseSameInLanguages should parse correctly in new mode: "!be+ cs# !ru+ uk+ pl*" 1`] = `"!be+ cs# pl* !ru+ uk+"`;

exports[`parseSameInLanguages should parse correctly in new mode: "!bg+ cz~ mk+" 1`] = `"!bg+ cs~ mk+"`;

exports[`parseSameInLanguages should parse correctly in new mode: "!v+ z#" 1`] = `"!be+ cs# pl# !ru+ sk# !uk+"`;

exports[`parseSameInLanguages should parse correctly in new mode: "" 1`] = `""`;

exports[`parseSameInLanguages should parse correctly in new mode: "sh* sl+" 1`] = `"bs* cnr* hr* sl+ sr*"`;
